apps/wallet-mobile/src/features/Receive/common/useAddressDerivationManager.tsx

```diff
@@ -0,0 +1,90 @@
+import {parseSafe, useAsyncStorage, useMutationWithInvalidations} from '@yoroi/common'
+import _ from 'lodash'
+import {UseMutationOptions, useQuery} from 'react-query'
+
+import {useSelectedWallet} from '../../../SelectedWallet'
+
+const keyAddressDerivationType = 'addressDerivationType'
+
+const useAddressDerivationMode = () => {
+  const storage = useAsyncStorage()
+  const wallet = useSelectedWallet()
+  const walletStorage = storage.join(`wallet/${wallet.id}/`)
+
+  const query = useQuery<AddressDerivation, Error>({
+    queryKey: [wallet.id, keyAddressDerivationType],
+    queryFn: async () => {
+      const addressDerivationType = await walletStorage.getItem(keyAddressDerivationType, parseAddressDerivation)
+
+      return addressDerivationType ?? initialAddressDerivation
+    },
+    suspense: true,
+  })
+
+  if (query.data == null) throw new Error('Invalid state')
+
+  return query.data
+}
+
+const useSetAddressDerivationMode = (options?: UseMutationOptions<void, Error, AddressDerivation>) => {
+  const storage = useAsyncStorage()
+  const wallet = useSelectedWallet()
+  const walletStorage = storage.join(`wallet/${wallet.id}/`)
+
+  const mutation = useMutationWithInvalidations({
+    mutationFn: (addressDerivationMode) => walletStorage.setItem(keyAddressDerivationType, addressDerivationMode),
+    invalidateQueries: [[wallet.id, keyAddressDerivationType]],
+    ...options,
+  })
+
+  return mutation.mutate
+}
+
+const useToogleAddressDerivationMode = ({...options}: UseMutationOptions<void, Error, void> = {}) => {
+  const storage = useAsyncStorage()
+  const wallet = useSelectedWallet()
+  const walletStorage = storage.join(`wallet/${wallet.id}/`)
+
+  const addressDerivationMode = useAddressDerivationMode()
+
+  const mutation = useMutationWithInvalidations({
+    mutationFn: () =>
+      walletStorage.setItem(keyAddressDerivationType, addressDerivationMode === 'single' ? 'multiple' : 'single'),
+    invalidateQueries: [[wallet.id, keyAddressDerivationType]],
+    ...options,
+  })
+
+  return {
+    toggle: mutation.mutate,
+    isToggleLoading: mutation.isLoading,
+  }
+}
+
+export type AddressDerivation = 'single' | 'multiple'
+const initialAddressDerivation: AddressDerivation = 'multiple'
```

Comment:
it should start with multiple, there is a need for the create wallet funnel to set it to single, old wallets and restore funnel will use the multiple as default cc: @SorinC6